feat(plugin-essentials): upgrade all packages with wildcard

Gelio opened the pull request:
**What's the problem this PR addresses?**

`yarn up '*'` only upgraded unscoped packages. I had to remember to run `yarn up '*/*'` later to also upgrade scoped packages. I was running these commands with the `-i` flag.

This PR changes it so `yarn up -i '*'` upgrades both scoped and unscoped packages.

Closes https://github.com/yarnpkg/berry/issues/3550

**How did you fix it?**

Use the [`basename`](https://www.npmjs.com/package/micromatch#user-content-optionsbasename) `micromatch` option so `*` matches scoped (e.g. `@types/node`) and unscoped (e.g. `react`) packages.

...

**Checklist**
<!--- Don't worry if you miss something, chores are automatically tested. -->
<!--- This checklist exists to help you remember doing the chores when you submit a PR. -->
<!--- Put an `x` in all the boxes that apply. -->
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

<!-- See https://yarnpkg.com/advanced/contributing#preparing-your-pr-to-be-released for more details. -->
<!-- Check with `yarn version check` and fix with `yarn version check -i` -->
- [x] I have set the packages that need to be released for my changes to be effective.
 
    Although I am not sure I did it right.

<!-- The "Testing chores" workflow validates that your PR follows our guidelines. -->
<!-- If it doesn't pass, click on it to see details as to what your PR might be missing. -->
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

merceyz commented on .yarn/versions/0e4a4eac.yml:
```diff
@@ -0,0 +1,4 @@
+releases:
+  "@yarnpkg/cli": minor
+  "@yarnpkg/plugin-essentials": minor
```
This is a breaking change so these need to be marked as `major`.

Gelio replied:
Do you think all of these should be `major` changes? Technically this could also be thought as a breaking change for the CLI, although only for interactive mode, which users review manually anyway

Gelio replied:
I marked this one as a `major` version bump. I left other ones intact. I also declined some other version bumps. I have no clue if this is correct since I don't know the inter-dependencies between these packages. Feel free to update this file yourself or let me know how it should be configured.

Gelio commented:
I removed the integration test that I added since this change only affects interactive mode (`-i`) and I didn't find an easy way to run yarn commands in interactive mode in the integration tests.

arcanis commented:
I added back the code that makes `yarn up '*'` upgrade everything, as I think it's best to keep the logic similar if possible between the interactive and non-interactive filtering. It makes the diff a major for the CLI but it's not a problem, since we want to release the 4.0 soon.
